KellyRyanGlobal/testing-app
# Kelly 15 multi container compose

PRe req
-Setup Jenkins in accordance with the test plan

Docker Desktop is installed on Windows with docker cli working
or
Docker is installed on centos
Docker-compose is installed locally (should be installed with docker desktop) https://docs.docker.com/compose/install/#install-compose

Test 1
1. Open a **windows terminal** and pull `kelly-15-multi-container-compose` into local
2. Run the instructions in the `README.md` under **Build Logger using docker-compose**
3. Run it the first time using ubuntu
4. Verify Output as written in the README
5. Open Docker Desktop and click Containers
6. Click **ubuntu** dropdown and then click **server_ubuntu**
7. 
![image](https://user-images.githubusercontent.com/89222071/140179642-7b34466f-141b-4cc1-947b-5c174ad11b30.png)
8. Verify the console output
9. 
![image](https://user-images.githubusercontent.com/89222071/140179741-1cd49833-1abd-46a6-9c11-490e3e68511c.png)


6. Repeat steps 3 and 9 for alpine


Test 2

1.  Log into jenkins and click the Logger-Pipeline-ubuntu job
2. Click configure
3.  Navigate to the **Add Parameter** and click **String Parameter**
4.  Enter the followingin the **Name** and **Default Value**
5. 
6. ![image](https://user-images.githubusercontent.com/89222071/139502943-a5fb6310-30c9-4b79-8ad4-e44ee748520b.png)
7. 
8.  Navigate down to the Branches to build field
9.  Enter `*/kelly-15-multi-container-compose`
10 Save the configuration
11. Click Build with parameters
12.  Click Build
13. Verify all the tests Pass
14. Repeat steps 3-13 for alpine
